src/CodeGen_Hexagon.cpp

```diff
@@ -1623,8 +1623,9 @@ Value *CodeGen_Hexagon::shuffle_vectors(Value *a, Value *b,
                 native_ty, MAKE_ID_PAIR(Intrinsic::hexagon_V6_hi).get(is_128B), {a});
             a = call_intrin_cast(
                 native_ty, MAKE_ID_PAIR(Intrinsic::hexagon_V6_lo).get(is_128B), {a});
-            a_ty = a->getType();
-            b_ty = b->getType();
+            a_ty = cast<llvm::VectorType>(a->getType());
```

Comment:
I'm confused. a_ty is an llvm::Type * local variable. Why would it be necessary to cast it the temporary to VectorType?

Reply:
In our local LLVM, this fails to compile with `incompatible pointer types assigning to 'llvm::VectorType *' from 'llvm::Type *'`... but yeah, something is definitely wonky here. Let me figure out what is up internally.